src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy

```diff
@@ -151,6 +152,80 @@ sourceSets {
         buildResult.output.contains(':checkScalafixIntegTest SKIPPED')
     }
 
+    def 'scalafix semantic rule RemoveUnused fixes scala code'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+
+        when:
+        runGradleTask('scalafix', [ ])
+
+        then:
+        scalaSrcFile.getText() == ScalafixTestConstants.FIXED_SCALA_CODE
+    }
+
+    def 'checkScalafix semantic rule RemoveUnused fails a build'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+
+        when:
+        runGradleTask('checkScalafix', [ ])
+
+        then:
+        // code is not changed on disk
+        scalaSrcFile.getText() == ScalafixTestConstants.FLAWED_SCALA_CODE
+        final UnexpectedBuildFailure err = thrown()
+        err.message.contains("A file on disk does not match the file contents if it was fixed with Scalafix")
+    }
+
+
+    def 'scalafix semantic rule RemoveUnused skips excluded scala files'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+        buildFile.append '''
+scalafix {
+  excludes = ["**/scalafix/**"]
+}
+'''
+        when:
+        runGradleTask('scalafix', [ ])
+
+        then:
+        scalaSrcFile.getText() == ScalafixTestConstants.FLAWED_SCALA_CODE
+    }
+
+    def 'scalafix syntactic rule DisableSyntax.noVars fails a build'() {
```

Comment:
- sug.: 'scalafix should run syntactic linter rule and fails the build if any violation is reported'
- it would be nice to set `autoConfigureSemanticdb` to false in tests where we run only syntactic rules